vslib/src/sai_vs_hostintf.cpp

```diff
@@ -36,10 +37,278 @@ typedef struct _hostif_info_t
 
     std::string name;
 
+    sai_object_id_t portid;
+
 } hostif_info_t;
 
 std::map<std::string, std::shared_ptr<hostif_info_t>> hostif_info_map;
 
+std::set<fdb_info_t> g_fdb_info_set;
+
+void processFdbInfo(
+        _In_ const fdb_info_t &fi,
+        _In_ sai_fdb_event_t fdb_event)
+{
+    SWSS_LOG_ENTER();
+
+    sai_attribute_t attrs[2];
+
+    attrs[0].id = SAI_FDB_ENTRY_ATTR_TYPE;
+    attrs[0].value.s32 = SAI_FDB_ENTRY_TYPE_DYNAMIC;
+
+    attrs[1].id = SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID;
+    attrs[1].value.oid = fi.bridge_port_id;
+
+    sai_fdb_event_notification_data_t data;
+
+    data.event_type = fdb_event;
+
+    data.fdb_entry = fi.fdb_entry;
+
+    if (data.fdb_entry.bridge_type == SAI_FDB_ENTRY_BRIDGE_TYPE_1Q)
+    {
+        // since this is only valid for 1D
+        data.fdb_entry.bridge_id = SAI_NULL_OBJECT_ID;
+    }
+    else if (data.fdb_entry.bridge_type == SAI_FDB_ENTRY_BRIDGE_TYPE_1D)
+    {
+        // since this is only valid for 1Q
+        data.fdb_entry.vlan_id = 0;
+    }
+    else
+    {
+        SWSS_LOG_ERROR("unknown bridge type %d", data.fdb_entry.bridge_type);
+    }
+
+    data.attr_count = 2;
+    data.attr = attrs;
+
+    // update metadata DB
+    meta_sai_on_fdb_event(1, &data);
+
+    sai_attribute_t attr;
+
+    attr.id = SAI_SWITCH_ATTR_FDB_EVENT_NOTIFY;
+
+    sai_status_t status = vs_generic_get(SAI_OBJECT_TYPE_SWITCH, data.fdb_entry.switch_id, 1, &attr);
+
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        return;
+    }
+
+    std::string s = sai_serialize_fdb_event_ntf(1, &data);
+
+    SWSS_LOG_DEBUG("calling user fdb event callback: %s", s.c_str());
+
+    sai_fdb_event_notification_fn ntf = (sai_fdb_event_notification_fn)attr.value.ptr;
+
+    if (ntf != NULL)
+    {
+        ntf(1, &data);
+    }
+}
+
+void findBridgeForPort(
```

Comment:
>findBridgeForPort [](start = 5, length = 17)

The bridge port lookup process is two steps:

1. use (vlan_id, phyiscal port_id) to match any .1D bridge port created. If there is match, then quit, found=true

2. use (physical port_id) to match any .1Q bridge created. if there is a match, the quite, found=true.

if found==true, generate fdb learn event on the .1D or .1Q bridge port. 

if not found, then do not generate fdb event. It means the packet is not received on the bridge port.

Reply:
is there dfault 1D bridge created on the switch ? because from what i noticed and what i implemented in current vs is that there is default 1Q bridge, so for example for TOR usage we will need to create 1D brige and move ports from 1Q to 1D bridge? and create bridge ports for that ?

Reply:
no, there is no default 1d bridge created on the switch.

Reply:
as we discussed on the im, im leaving this and just looking for any bridge since we will remove bridge when using router interface, and i put info in to the comment